### PR TITLE
Merge 3.X into Master

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

**Comment by VincentLanglet** [comment]
I don't know if we should (keep) introduce phpdoc in 3.x, the conflict on master are huge. @sonata-project/contributors @core23

File: src/Admin/AbstractAdmin.php [hunk 1]
```diff
@@ -1760,7 +1776,10 @@ public function isCurrentChild(): bool
     public function getCurrentChildAdmin(): ?AdminInterface
     {
         foreach ($this->children as $children) {
-            if ($children->isCurrentChild()) {
+            // NEXT_MAJOR: Remove method_exists check and delete elseif case
```
Comment (core23) [inline src/Admin/AbstractAdmin.php:1779]: You can revert this. This is already resolved for the master branch.

File: src/Admin/AdminInterface.php [hunk 2]
```diff
@@ -244,16 +265,19 @@ public function getUniqid(): string;
...
     /**
      * Returns the classname label.
-     *
-     * @return string the classname label
      */
     public function getClassnameLabel(): string;
 
     /**
-     * @param mixed $id
+     * @param mixed id
```
Comment (core23) [inline src/Admin/AdminInterface.php:272]:
```suggestion
     * @param mixed $id
```

File: src/Admin/AdminInterface.php [hunk 3]
```diff
@@ -349,18 +379,21 @@ public function setFormGroups(array $formGroups): void;
...
     /**
      * @return array<string, mixed>
      */
     public function getShowTabs(): array;
 
-    public function setShowTabs(array $showTabs): void;
-
     /**
-     * Remove a form group field.
+     * @param array<string, mixed> $formTabs
```
Comment (core23) [inline src/Admin/AdminInterface.php:393]:
```suggestion
     * @param array<string, mixed> $showTabs
```

**Review by core23 — CHANGES_REQUESTED** [review]

File: src/Admin/BreadcrumbsBuilder.php [hunk 4]
```diff
@@ -46,7 +46,7 @@ public function configureOptions(OptionsResolver $resolver): void
         ]);
     }
 
-    public function getBreadcrumbs(AdminInterface $admin, string $action): iterable
+    public function getBreadcrumbs(AdminInterface $admin, string $action): array
```
Comment (jordisala1991) [inline src/Admin/BreadcrumbsBuilder.php:49]: this one does not seem right

Comment (VincentLanglet) [reply src/Admin/BreadcrumbsBuilder.php:49]: See https://github.com/sonata-project/SonataAdminBundle/pull/6379/files#diff-d034497868502fe990e3db10a7c2eefbR33

Type was changed in 3.x from iterable to array.
So, return typehint has to follow.

If you want to rollback this change, it should start with 3.x and not block this merge.

Comment (jordisala1991) [reply src/Admin/BreadcrumbsBuilder.php:49]: Thats the problem changing phpdoc without real testing those types, lets hope we dont have to deal with a lot more prs on 3.x about that

**Review by jordisala1991 — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Please take a new look @core23  :)

**Review by jordisala1991 — APPROVED** [review]

**Review by core23 — APPROVED** [review]

**Pull request merged**